app/controllers/smart_proxies_controller.rb

```diff
@@ -64,6 +64,7 @@ def puppet_dashboard
     dashboard = Dashboard::Data.new("puppetmaster = \"#{@smart_proxy.name}\"")
     @hosts = dashboard.hosts
     @report = dashboard.report
+    @latest_events = dashboard.latest_events
```

Comment:
How about extracting these 4 lines to a concern with a method that takes the filter as a parameter?